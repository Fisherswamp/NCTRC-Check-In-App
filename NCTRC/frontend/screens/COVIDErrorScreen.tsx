import React from 'react';
import { SafeAreaView, StyleSheet, Text, View, Button, Image } from 'react-native';
import { StackNavigationProp } from '@react-navigation/stack';
import { AuthStackParamList, AppScreens } from '../index';
type CovidErrorScreenNavigationProps = StackNavigationProp<AuthStackParamList, AppScreens.CovidError>;
interface CovidErrorScreenProps {
    navigation: CovidErrorScreenNavigationProps;
    route: { params: ErrorParams };
}

export type ErrorParams = {
    name: string;
    email: string;
    reason: string;
    concerns: string;
};

const styles = StyleSheet.create({});
const CovidErrorScreen: React.FunctionComponent<CovidErrorScreenProps> = (props) => {
    const { navigation, route } = props;
    const { params } = route;
    const { name, email, reason, concerns } = params;
    return (
        <SafeAreaView>
            <View>
                <Image source={require('./../assets/NCTRClogo.png')} style={{ width: 400, height: 400 }}></Image>
                <Text>Sorry, we are not able to let you in the farm because you indicated {reason}</Text>
                <Button color="#884633" title="Back" onPress={() => navigation.pop()} />
                <Button color="#884633" title="Home" onPress={() => navigation.popToTop()} />
            </View>
        </SafeAreaView>
    );
};
export default CovidErrorScreen;
