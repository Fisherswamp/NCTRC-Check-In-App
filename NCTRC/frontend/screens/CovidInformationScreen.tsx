import React, { useState, SetStateAction } from 'react';
import { SafeAreaView, StyleSheet, Text, View, Button, Image, TextInput, Picker } from 'react-native';
import { StackNavigationProp } from '@react-navigation/stack';
import { AuthStackParamList, AppScreens } from '../Index';
import { ScrollView } from 'react-native-gesture-handler';
import { setStatusBarNetworkActivityIndicatorVisible } from 'expo-status-bar';
type CovidInformationScreenNavigationProps = StackNavigationProp<AuthStackParamList, AppScreens.CovidInformation>;
export type InfoParams = {
    name: string;
    email: string;
};

interface CovidInformationScreenProps {
    route: { params: InfoParams };
    navigation: CovidInformationScreenNavigationProps;
}

const styles = StyleSheet.create({});
const maxCap = 10;
const currentCap = 10;

const CovidInformationScreen: React.FunctionComponent<CovidInformationScreenProps> = (props) => {
    const { navigation, route } = props;
    const { params } = route;
    const { name, email } = params;
    const [traveled, setTraveled] = React.useState(false);
    const [concerns, setConcerns] = useState('');
    const [contact, setContact] = React.useState(false);
    const [gatherings, setGatherings] = React.useState(false);
    const [isolate, setIsolate] = React.useState(false);
    const [travelLabel, setravelLabel] = useState('');
    const [contactLabel, setContactLabel] = useState('');
    const [gatheringsLabel, setGatheringsLabel] = useState('');
    const [isolateLabel, setIsolateLabel] = useState('');
    const [tempurature, setTempurature] = useState('');

    return (
        <ScrollView>
            <View>
                <Image source={require('./../assets/NCTRClogo.png')} style={{ width: 10, height: 10 }}></Image>
                <Text>Name</Text>
                <TextInput placeholder={name} editable={false} />
                <Text>Email</Text>
                <TextInput placeholder={email} editable={false} />
                <Text>
                    Within the past 24 hours have you or anyone in your household experienced any of the following
                    symptoms: fever (over 100.4 F), sore or dry throat, shortness of breath, chest congestion, cough,
                    loss of taste or sense of smell, sneezing, and/or rash?
                </Text>
                <Picker onValueChange={(itemValue) => setContact(itemValue)}>
                    <Picker.Item value="" label="" />
                    <Picker.Item label="Yes" value={true} />
                    <Picker.Item label="No" value={false} />
                </Picker>
                <Text>Have you traveled out of the country or out of the state within the past 14 days?</Text>
                <Picker onValueChange={(itemValue) => setTraveled(itemValue)}>
                    <Picker.Item value="" label="" />
                    <Picker.Item label="Yes" value={true} />
                    <Picker.Item label="No" value={false} />
                </Picker>
                <Text>
                    Have you attended an event or gathering of more than 25 people at any point in the last 14 days?
                </Text>
                <Picker onValueChange={(itemValue) => setGatherings(itemValue)}>
                    <Picker.Item value="" label="" />
                    <Picker.Item label="Yes" value={true} />
                    <Picker.Item label="No" value={false} />
                </Picker>
                <Text>
                    Within the past 14 days, have you received a positive test result for COVID-19 or are you awaiting
                    test results for COVID-19?
                </Text>
                <Picker onValueChange={(itemValue) => setIsolate(itemValue)}>
                    <Picker.Item value="" label="" />
                    <Picker.Item label="Yes" value={true} />
                    <Picker.Item label="No" value={false} />
                </Picker>
                <Text>Tempurature</Text>
                <TextInput onChangeText={(text) => setTempurature(text)} />
                <Text>Is there anything else you would like to share? Questions, concerns, etc.</Text>
                <TextInput onChangeText={(text) => setConcerns(text)} />
                <Button
                    color="#884633"
                    title="Submit"
                    onPress={() => {
                        if (currentCap + 1 > maxCap) {
                            navigation.navigate(AppScreens.MaxCap);
                        } else if (traveled) {
                            setIsolate(false);
                            setGatherings(false);
                            setTraveled(false);
                            setContact(false);
                            navigation.navigate(AppScreens.CovidError, {
                                name: name,
                                email: email,
                                reason: 'Traveled in the last 30 days',
                                concerns: concerns,
                            });
                        } else if (contact) {
                            navigation.navigate(AppScreens.CovidError, {
                                name: name,
                                email: email,
                                reason: 'Contact with person who tested positive for COVID-19',
                                concerns: concerns,
                            });
                        } else if (gatherings) {
                            navigation.navigate(AppScreens.CovidError, {
                                name: name,
                                email: email,
                                reason: 'Attended large gathering',
                                concerns: concerns,
                            });
                        } else if (isolate) {
                            navigation.navigate(AppScreens.CovidError, {
                                name: name,
                                email: email,
                                reason: 'Asked to self isolate',
                                concerns: concerns,
                            });
                        } else {
                            navigation.navigate(AppScreens.SignInLanding, {
                                name: name,
                                email: email,
                                traveled: traveled,
                                contact: contact,
                                concerns: concerns,
                                gatherings: gatherings,
                                isolate: isolate,
                            });
                        }
                    }}
                />
                <Button color="#884633" title="Back" onPress={() => navigation.pop()} />
                <Button color="#884633" title="Home" onPress={() => navigation.popToTop()} />
            </View>
        </ScrollView>
    );
};

export default CovidInformationScreen;
