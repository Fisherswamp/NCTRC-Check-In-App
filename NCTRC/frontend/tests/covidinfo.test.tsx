
import React from "react";
import  CovidInformationScreen from '../screens/CovidInformationScreen';
import  COVIDErrorScreen from '../screens/COVIDErrorScreen';
import  ErrorParams from '../screens/COVIDErrorScreen';
import { shallow, configure } from 'enzyme'; 
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const props = {
    navigation: {
        addListener: jest.fn(), 
        canGoBack: jest.fn(), 
        dangerouslyGetParent: jest.fn(), 
        dangerouslyGetState: jest.fn(), 
        dispatch: jest.fn(), 
        goBack: jest.fn(), 
        isFocused: jest.fn(), 
        navigate: jest.fn(), 
        pop: jest.fn(), 
        popToTop: jest.fn(), 
        push: jest.fn(), 
        removeListener: jest.fn(), 
        replace: jest.fn(), 
        reset: jest.fn(), 
        setOptions: jest.fn(), 
        setParams: jest.fn(), 
    }, 
    route: {
        params: {
            firstName: "string",
            lastName: "string",
            email: "string",
        }
    },  
    errorRoute: {
        params: {
            reason: "string",
        }

    }
}


describe('CovidInformationScreen renders', () => {
    it('CovidInformationScreen renders', () => {
      const tree = renderer.create(<CovidInformationScreen navigation={props.navigation} route={props.route} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

describe('COVIDErrorScreen renders', () => {
    it('COVIDErrorScreen render', () => {
      const tree = renderer.create(<COVIDErrorScreen navigation={props.navigation} route={props.errorRoute} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});


describe('CovidInformationScreen elements', () => {
    it("CovidInformationScreen elements", () => {
       let wrapper =  shallow(<CovidInformationScreen navigation={props.navigation} route={props.route} />);
       expect(wrapper.find('Image').length).toBe(2); 
       expect(wrapper.find('Text').length).toBe(18); 
       expect(wrapper.find('View').length).toBe(6); 
       expect(wrapper.find('ScrollView').length).toBe(1); 

    })
   
});

describe('COVIDErrorScreen elements', () => {
    it("COVIDErrorScreen elements", () => {
       let wrapper =  shallow(<COVIDErrorScreen navigation={props.navigation} route={props.errorRoute} />);
       expect(wrapper.find('Image').length).toBe(2); 
       expect(wrapper.find('Text').length).toBe(2); 
       expect(wrapper.find('View').length).toBe(2); 

    })
   
});
jest.useFakeTimers(); 