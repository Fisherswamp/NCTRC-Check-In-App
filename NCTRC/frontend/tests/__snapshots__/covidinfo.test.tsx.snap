// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COVIDErrorScreen renders COVIDErrorScreen render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/backbutton.png",
        }
      }
      style={
        Object {
          "height": 75,
          "width": 75,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "marginTop": -40,
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/NCTRClogo.png",
        }
      }
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#884633",
          "borderRadius": 9,
          "borderWidth": 1,
          "marginTop": -150,
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#884633",
            "fontSize": 15,
            "margin": 40,
          }
        }
      >
        Sorry, we are not able to let you in the farm because 
        string
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#884633",
            "borderRadius": 9,
            "height": 60,
            "margin": 20,
            "marginBottom": 50,
            "opacity": 1,
            "width": 180,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "marginTop": 18,
            }
          }
        >
          Home
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`CovidInformationScreen renders CovidInformationScreen renders 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "marginTop": 40,
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "flex-start",
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/backbutton.png",
          }
        }
        style={
          Object {
            "height": 75,
            "width": 75,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "marginTop": -40,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/NCTRClogo.png",
          }
        }
        style={
          Object {
            "height": 150,
            "width": 150,
          }
        }
      />
    </View>
    <View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 15,
            "margin": 10,
            "marginLeft": 20,
            "width": 300,
          }
        }
      >
        First Name
      </Text>
      <TextInput
        allowFontScaling={true}
        editable={false}
        placeholder="string"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#884633",
            "borderRadius": 9,
            "borderWidth": 1,
            "height": 60,
            "margin": 10,
            "paddingHorizontal": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 15,
            "margin": 10,
            "marginLeft": 20,
            "width": 300,
          }
        }
      >
        Last Name
      </Text>
      <TextInput
        allowFontScaling={true}
        editable={false}
        placeholder="string"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#884633",
            "borderRadius": 9,
            "borderWidth": 1,
            "height": 60,
            "margin": 10,
            "paddingHorizontal": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 15,
            "margin": 10,
            "marginLeft": 20,
            "width": 300,
          }
        }
      >
        Email
      </Text>
      <TextInput
        allowFontScaling={true}
        editable={false}
        placeholder="string"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#884633",
            "borderRadius": 9,
            "borderWidth": 1,
            "height": 60,
            "margin": 10,
            "paddingHorizontal": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 15,
            "margin": 10,
            "marginLeft": 20,
            "width": 300,
          }
        }
      >
        Within the past 24 hours have you or anyone in your household experienced any of the following symptoms: fever (over 100.4 F), sore or dry throat, shortness of breath, chest congestion, cough, loss of taste or sense of smell, sneezing, and/or rash?
      </Text>
      <View
        accessibilityRole="radiogroup"
        accessible={true}
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#884633",
                "fontSize": 15,
                "padding": 5,
              }
            }
          >
            Yes
          </Text>
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              Object {
                "checked": false,
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "height": 20,
                  "margin": 8,
                  "width": 20,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "color": "#884633",
                "fontSize": 15,
                "padding": 5,
              }
            }
          >
            No
          </Text>
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              Object {
                "checked": false,
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "height": 20,
                  "margin": 8,
                  "width": 20,
                }
              }
            />
          </View>
        </View>
      </View>
      <View>
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "grey",
              "fontSize": 15,
              "margin": 10,
              "marginLeft": 20,
              "width": 300,
            }
          }
        >
          Have you traveled out of the country or out of the state within the past 14 days?
        </Text>
        <View
          accessibilityRole="radiogroup"
          accessible={true}
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#884633",
                  "fontSize": 15,
                  "padding": 5,
                }
              }
            >
              Yes
            </Text>
            <View
              accessibilityLiveRegion="polite"
              accessibilityRole="radio"
              accessibilityState={
                Object {
                  "checked": false,
                  "disabled": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 18,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "borderColor": "rgba(0, 0, 0, 0.54)",
                    "borderRadius": 10,
                    "borderWidth": 2,
                    "height": 20,
                    "margin": 8,
                    "width": 20,
                  }
                }
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#884633",
                  "fontSize": 15,
                  "padding": 5,
                }
              }
            >
              No
            </Text>
            <View
              accessibilityLiveRegion="polite"
              accessibilityRole="radio"
              accessibilityState={
                Object {
                  "checked": false,
                  "disabled": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 18,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "borderColor": "rgba(0, 0, 0, 0.54)",
                    "borderRadius": 10,
                    "borderWidth": 2,
                    "height": 20,
                    "margin": 8,
                    "width": 20,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 15,
            "margin": 10,
            "marginLeft": 20,
            "width": 300,
          }
        }
      >
        Have you attended an event or gathering of more than 25 people at any point in the last 14 days?
      </Text>
      <View
        accessibilityRole="radiogroup"
        accessible={true}
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#884633",
                "fontSize": 15,
                "padding": 5,
              }
            }
          >
            Yes
          </Text>
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              Object {
                "checked": false,
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "height": 20,
                  "margin": 8,
                  "width": 20,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "color": "#884633",
                "fontSize": 15,
                "padding": 5,
              }
            }
          >
            No
          </Text>
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              Object {
                "checked": false,
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "height": 20,
                  "margin": 8,
                  "width": 20,
                }
              }
            />
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 15,
            "margin": 10,
            "marginLeft": 20,
            "width": 300,
          }
        }
      >
        Within the past 14 days, have you received a positive test result for COVID-19 or are you awaiting test results for COVID-19?
      </Text>
      <View
        accessibilityRole="radiogroup"
        accessible={true}
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#884633",
                "fontSize": 15,
                "padding": 5,
              }
            }
          >
            Yes
          </Text>
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              Object {
                "checked": false,
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "height": 20,
                  "margin": 8,
                  "width": 20,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "color": "#884633",
                "fontSize": 15,
                "padding": 5,
              }
            }
          >
            No
          </Text>
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              Object {
                "checked": false,
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 18,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 10,
                  "borderWidth": 2,
                  "height": 20,
                  "margin": 8,
                  "width": 20,
                }
              }
            />
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 15,
            "margin": 10,
            "marginLeft": 20,
            "width": 300,
          }
        }
      >
        What is your tempurature?
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#884633",
            "borderRadius": 9,
            "borderWidth": 1,
            "height": 60,
            "margin": 10,
            "paddingHorizontal": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 15,
            "margin": 10,
            "marginLeft": 20,
            "width": 300,
          }
        }
      >
        Is there anything else you would like to share? Questions, concerns, etc.
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#884633",
            "borderRadius": 9,
            "borderWidth": 1,
            "height": 60,
            "margin": 10,
            "paddingHorizontal": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#884633",
            "borderRadius": 9,
            "height": 60,
            "margin": 20,
            "marginBottom": 50,
            "opacity": 1,
            "width": 180,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "marginTop": 18,
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
